### ISSUE TITLE
liana-plus misty and mistyR importance interpretation

### ISSUE THREAD

**Issue by asmlgkj** [saezlab/liana-py#209]
Dear liana+ Development Team,
Thank you for developing MISTyR - it's an excellent tool that has greatly advanced our understanding of spatial cellular interactions in tissue architecture.
I have a technical question about feature importance calculation that I hope you can clarify:
Main Question:
When using MISTyR with the default random forest model, I'm observing negative importance values in the interaction heatmaps. However, I understood that standard random forest importance (mean decrease in Gini impurity) should always be non-negative.
Specific observations:
r# Using default settings
run_misty(heart_views, "results")
misty_results <- collect_results("results")
misty_results %>% plot_interaction_heatmap(view = "intra", clean = TRUE)

Shows negative values in the heatmap
Comparison with LIANA-plus:
I noticed that LIANA-plus explicitly distinguishes between:

Random Forest model: Uses "mean decrease in Gini impurity" (≥0)
Linear model: Uses "t-values of model coefficients" (can be negative)

Their documentation specifically mentions using t-values because "coefficients would not be directly comparable (as are importances in a Random Forest)."
My questions:

Does MISTyR use permutation importance instead of Gini impurity for random forests? This would explain the negative values.
What is the biological interpretation of negative importance values?

Should they be interpreted as spatial exclusion/competition between cell types?
Or as confounding/noise that hurts prediction?

How does MISTyR's approach differ from LIANA-plus in terms of importance calculation and interpretation?

**Comment by asmlgkj** [comment]
Thanks a lot
I ran the tutorial/example dataset from the docs end-to-end to compute cell–cell co-localization/interaction features and then compared LIANA-plus importance estimates using:

a random forest model (default settings), and

a linear model (default settings; t-values for signed effects),

and I also checked MISTyR (variance explained per feature).

What I observe

The linear model yields importances/t-values that are mostly close to zero—suggesting little strong co-localization or exclusion.

The random forest shows clearer signals for some pairs (higher importances).

MISTyR gives yet another ranking/pattern (variance explained), not matching either RF or the linear model.

This leaves me uncertain about which estimator to trust and how to interpret the differences.

My questions

Model choice: In your experience, when would you recommend RF over a linear model, and vice-versa, for neighborhood/co-localization signals? Is RF mostly capturing non-linearities/interactions that the linear model misses, or could it be overfitting/picking up noise?

Normalization/comparability:

LIANA-plus RF importances are reported as fractions of variance explained per target (sum to 1), whereas linear models give signed t-values and MISTyR reports raw variance explained.

Do you suggest a common normalization (e.g., per-target z-scores, permutation importances) to make these more comparable across methods?

Signed effects & negatives: For the linear model, near-zero or small t-values could reflect multicollinearity, weak signal, or confounding. Are there recommended pre-checks (feature scaling, VIF, residual diagnostics) or regularization (e.g., ridge/LASSO) you advise before concluding “no effect”?

Stability & validation: Any best practices for CV/bootstrapping, random seeds, or permutation testing to assess the stability of importances across RF/linear/MISTyR? Do you have default hyperparameter settings you recommend for RF (n_estimators, max_depth, mtry) to reduce variance without washing out true non-linear signal?

Interpretation guide: When RF highlights pairs but the linear model is flat (and MISTyR differs again), should we interpret RF signals as non-linear relationships (biologically meaningful) or as potential artifacts unless they replicate across models? Any decision rule you recommend (e.g., require concordance across two estimators; or accept RF if permutation importance is stable)?

Confounders/covariates: For neighborhood models, do you recommend including known covariates (library size, tissue region, batch, total neighbors, distance metrics) explicitly in the linear model and as stratification in RF, to reduce spurious co-localization?
